--- ./deconz-rest-plugin/de_web.pro	2017-03-04 23:50:14.933712433 +0100
+++ ./de_web.pro	2017-02-27 22:34:59.256388661 +0100
@@ -33,7 +33,7 @@
 }
 
 win32:LIBS +=  -L../.. -ldeCONZ1
-unix:LIBS +=  -L../.. -ldeCONZ
+unix:LIBS +=  -L../.. -ldeCONZ -lcrypto -lssl
 win32:CONFIG += dll
 
 unix:!macx {
@@ -91,6 +91,7 @@
            rest_node_base.h \
            rule.h \
            scene.h \
+           ct_push_bridge.h \
            sensor.h
 
 SOURCES  = authentification.cpp \
@@ -129,6 +130,7 @@
            sensor.cpp \
            reset_device.cpp \
            rest_userparameter.cpp \
+           ct_push_bridge.cpp \
            zcl_tasks.cpp
 
 win32:DESTDIR  = ../../debug/plugins # TODO adjust
