--- ./deconz-rest-plugin/de_web.pro	2017-02-18 00:29:09.617999771 +0100
+++ ./de_web.pro	2017-02-17 20:59:07.147149498 +0100
@@ -91,6 +91,7 @@
            rest_node_base.h \
            rule.h \
            scene.h \
+           ct_push_bridge.h \
            sensor.h
 
 SOURCES  = authentification.cpp \
@@ -129,6 +130,7 @@
            sensor.cpp \
            reset_device.cpp \
            rest_userparameter.cpp \
+           ct_push_bridge.cpp \
            zcl_tasks.cpp
 
 win32:DESTDIR  = ../../debug/plugins # TODO adjust
