--- ./deconz-rest-plugin/de_web_plugin.cpp	2016-12-28 21:57:34.207658398 +0100
+++ ./de_web_plugin.cpp	2016-12-28 15:06:35.921423767 +0100
@@ -486,6 +486,14 @@
         return;
     }
 
+#ifdef ENABLE_PUSH
+    if ( !sensor->m_pushDevice ) {
+        DEBUGFTRACE ( "gpProcessButtonEvent seq:" << (int) ind.frameCounter() );
+        
+        sensor->UpdateSensorFhem ();
+    }
+#endif			
+
     QString lastUpdatedOld = sensor->state().lastupdated();
 
     sensor->state().setButtonevent(ind.gpdCommandId());
@@ -551,7 +559,7 @@
                     }
                 }
             }
-
+        
             QString body = "";
             QString method = "";
             QStringList idList;
@@ -911,7 +919,7 @@
                 }
             }
         }
-    }
+    }	
 }
 
 /*! Returns the number of tasks for a specific address.
