--- ./deconz-rest-plugin/de_web_plugin.cpp	2017-02-18 00:29:09.617999771 +0100
+++ ./de_web_plugin.cpp	2017-02-17 20:59:07.157149448 +0100
@@ -504,6 +504,14 @@
         return;
     }
 
+#ifdef ENABLE_PUSH
+	if ( !sensor->m_pushDevice ) {
+		DEBUGFTRACE ( "gpProcessButtonEvent seq:" << ( int ) ind.frameCounter () );
+
+		sensor->UpdateSensorFhem ();
+	}
+#endif		
+
     QString lastUpdatedOld = sensor->state().lastupdated();
 
     sensor->state().setButtonevent(ind.gpdCommandId());
