--- ./deconz-rest-plugin/group.cpp	2017-02-18 00:29:09.617999771 +0100
+++ ./group.cpp	2017-02-17 20:59:07.157149448 +0100
@@ -10,6 +10,7 @@
 
 #include "group.h"
 #include <QStringList>
+#include "ct_push_bridge.h"
 
 /*! Constructor.
  */
@@ -43,6 +44,8 @@
  */
 void Group::setAddress(uint16_t address)
 {
+	pushSetGroupInfoCompName1 ( addr, address );
+
     m_addr = address;
     m_id = QString::number(address);
 }
@@ -95,6 +98,12 @@
 void Group::setIsOn(bool on)
 {
     m_on = on;
+
+	if ( on ) {
+		pushUpdateGroupInfo ();
+	}
+
+	pushBridge.SetGroupInfo ( m_name, m_id, "on", ( uint32_t ) ( on ? 1 : 0 ) );
 }
 
 /*! Sets the group color loop active state.
@@ -102,6 +111,8 @@
  */
 void Group::setColorLoopActive(bool colorLoopActive)
 {
+	pushSetGroupInfoComp ( colorLoopActive );
+
     m_colorLoopActive = colorLoopActive;
 }
 
@@ -177,6 +188,8 @@
     {
         m_deviceMemberships.push_back(*i);
     }
+
+	pushUpdateGroupInfo ();
 }
 
 /*! lightsequence to string. */
@@ -211,6 +224,8 @@
     {
         m_lightsequence.push_back(*i);
     }
+
+	pushUpdateGroupInfo ();
 }
 
 /*! Returns the scene for a given \p sceneId or 0 if not present. */
