--- ./deconz-rest-plugin/group.cpp	2017-03-04 23:50:14.983712234 +0100
+++ ./group.cpp	2017-02-27 01:18:12.342487470 +0100
@@ -10,6 +10,7 @@
 
 #include "group.h"
 #include <QStringList>
+#include "ct_push_bridge.h"
 
 /*! Constructor.
  */
@@ -43,6 +44,8 @@
  */
 void Group::setAddress(uint16_t address)
 {
+	pushSetGroupInfoCompName1 ( addr, (uint32_t) address );
+
     m_addr = address;
     m_id = QString::number(address);
 }
@@ -95,6 +98,13 @@
 void Group::setIsOn(bool on)
 {
     m_on = on;
+
+#ifdef ENABLE_PUSH
+	if ( on ) { pushUpdateGroupInfo (); }
+
+	pushBridge.SetGroupInfo ( m_id, "on", ( uint32_t ) ( on ? 1 : 0 ) );
+	pushBridge.SetGroupInfo ( m_id, "state", ( on ? "on" : "off" ) );
+#endif
 }
 
 /*! Sets the group color loop active state.
@@ -102,6 +112,8 @@
  */
 void Group::setColorLoopActive(bool colorLoopActive)
 {
+	pushSetGroupInfoCompUI32 ( colorLoopActive );
+
     m_colorLoopActive = colorLoopActive;
 }
 
@@ -177,6 +189,8 @@
     {
         m_deviceMemberships.push_back(*i);
     }
+
+	pushUpdateGroupInfo ();
 }
 
 /*! lightsequence to string. */
@@ -211,6 +225,8 @@
     {
         m_lightsequence.push_back(*i);
     }
+
+	pushUpdateGroupInfo ();
 }
 
 /*! Returns the scene for a given \p sceneId or 0 if not present. */
