--- ./deconz-rest-plugin/rest_node_base.h	2017-03-04 23:50:14.983712234 +0100
+++ ./rest_node_base.h	2017-02-17 20:59:07.157149448 +0100
@@ -13,6 +13,7 @@
 
 #include <QTime>
 #include <deconz.h>
+#include "ct_push_bridge.h"
 
 /*! \class NodeValue
 
@@ -77,6 +78,8 @@
     const NodeValue &getZclValue(quint16 clusterId, quint16 attributeId) const;
     NodeValue &getZclValue(quint16 clusterId, quint16 attributeId);
 
+	pushNodeExt ()
+
 private:
     deCONZ::Node *m_node;
     deCONZ::Address m_addr;
