--- ./deconz-rest-plugin/sensor.h	2017-03-04 23:50:14.983712234 +0100
+++ ./sensor.h	2017-02-17 20:59:07.167149398 +0100
@@ -46,6 +46,8 @@
     void setLux(quint32 lux);
     void updateTime();
 
+	pushClassSensorExt ()
+
 private:
     QString m_lastupdated;
     QString m_flag;//bool
@@ -86,6 +88,8 @@
     const QString &sunsetoffset() const;
     void setSunsetoffset(const QString &sunsetoffset);
 
+	pushClassSensorExt ()
+
 private:
     bool m_on;
     bool m_reachable;
@@ -176,6 +180,8 @@
     QVector<QString> sensorTypes;
     QString etag;
 
+	pushClassMethodSensor ()
+
 private:
     DeletedState m_deletedstate;
     QString m_name;
