--- ./deconz-rest-plugin/sensor.h	2016-12-28 21:57:34.207658398 +0100
+++ ./sensor.h	2016-12-28 16:45:04.080169286 +0100
@@ -45,6 +45,8 @@
     void setLux(quint32 lux);
     void updateTime();
 
+	pushClassSensorExt ()
+    
 private:
     QString m_lastupdated;
     QString m_flag;//bool
@@ -85,6 +87,8 @@
     const QString &sunsetoffset() const;
     void setSunsetoffset(const QString &sunsetoffset);
 
+	pushClassSensorExt ()
+
 private:
     bool m_on;
     bool m_reachable;
@@ -168,6 +172,8 @@
     QVector<QString> sensorTypes;
     QString etag;
 
+    pushClassMethodSensor ()
+
 private:
     DeletedState m_deletedstate;
     QString m_name;
